arches/app/media/js/viewmodels/reference-select.js

```diff
@@ -0,0 +1,158 @@
+                    const flatItems = self.flattenItems(items, [], []);
+                    flatItems.forEach(item => {
+                        item["listid"] = item.id;
+                        item.id = item.uri;
+                    });
+                    return {
+                        "results": flatItems,
+                        "pagination": {
+                            "more": false
+                        }
+                    };
+                }
+            },
+            templateResult: function(item) {
+                let indentation = '';
+                for (let i = 0; i < item.depth; i++) {
+                    indentation += '&nbsp;&nbsp;&nbsp;&nbsp;';
+                }
+
+                if (item.uri) {
+                    const text = indentation + self.getPrefLabel(item.labels) || arches.translations.searching + '...';
+                    NAME_LOOKUP[item.uri] = {"prefLabel": text, "labels": item.labels, "listid": item.listid};
+                    return text;
+                }
+            },
+            templateSelection: function(item) {
+                if (!item.uri) { // option has a different shape when coming from initSelection vs templateResult
+                    return item.text; 
+                } else {
+                    return NAME_LOOKUP[item.uri]["prefLabel"];
+                }
+            },
+            escapeMarkup: function(m) { return m; },
+            initComplete: false,
+            initSelection: function(el, callback) {
+
+                const setSelectionData = function(data) {
+                    const valueData = koMapping.toJS(self.value());
+                    valueData.forEach(function(value) {
+                        NAME_LOOKUP[value.uri] = {
+                                "prefLabel": self.getPrefLabel(value.labels),
+                                "labels": value.labels,
+                                "listid": value.listid 
+                            };
+                    });
+       
+                    if(!self.select2Config.initComplete){
+                        valueData.forEach(function(data) {
+                            const option = new Option(
+                                self.getPrefLabel(data.labels),
+                                data.uri,
+                                true, 
+                                true
+                            );
+                            $(el).append(option);
+                            self.selectionValue().push(data.uri);
+                        });
+                        self.select2Config.initComplete = true;
+                    }
+                    callback(valueData);
+                };
+
+                if (self.value()?.length) {
+                    setSelectionData();
+                } else {
+                    callback([]);
+                }
+
+            }
```

Comment:
In this scenario:
- node allows multiple references
- tile is saved with 2 references
- click to "X" the 2nd entry
- click "Cancel Edit"


The original entry is not restored.